tests/core/scheduler/test_scheduler.py

```diff
@@ -44,6 +50,14 @@ def reset_configuration_singleton():
         os.remove(f)
 
 
+@pytest.fixture(scope="function", autouse=True)
+def reset_scheduler_singleton():
+    yield
+    _Scheduler._set_nb_of_workers(None)
+    _Scheduler.jobs_to_run = Queue()
+    _Scheduler.blocked_jobs = []
```

Comment:
Could we put this code in the _conftest.py_ ? I think it's present multiple times in this PR.

Reply:
agree!